src/test/java/com/wojciechkocik/usage/service/TimeSpentCrossMidnightServiceTest.java

```diff
@@ -0,0 +1,143 @@
+package com.wojciechkocik.usage.service;
+
+import com.wojciechkocik.usage.dto.DailyUsage;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.data.util.Pair;
+
+import javax.persistence.Tuple;
+import java.time.Duration;
+import java.time.LocalTime;
+import java.time.ZonedDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * @author Wojciech Kocik
+ * @since 28.03.2017
+ */
+public class TimeSpentCrossMidnightServiceTest {
+
+    private TimeSpentCrossMidnightService timeSpentCrossMidnightService;
+
+    @Before
+    public void setUp() throws Exception {
+        timeSpentCrossMidnightService = new TimeSpentCrossMidnightServiceImpl();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    private ZonedDateTime getMidnightOfDate(ZonedDateTime dateTime) {
+        return dateTime
+                .toLocalDate().atTime(LocalTime.MIDNIGHT).atZone(dateTime.getZone());
+    }
+
+    @Test
+    public void shouldProperlyDivideCrossedMidnightSessionDozenDays() throws Exception {
+        //Arrange
+        ZonedDateTime timeDateStarted = ZonedDateTime.parse("2017-01-23T23:59:00.000+01:00[Europe/Warsaw]");
+        long sessionTime = 60 * 60 * 25 * 10;
+        long[] sessionTimeForEveryDay = {60, 86400, 86400, 86400, 86400, 86400, 86400, 86400, 86400, 86400, 86400, 35940};
+
+        //Act & Assert
+        shouldProperlyDivideCrossedMidnightSession_parametrized(timeDateStarted, sessionTime, sessionTimeForEveryDay);
+    }
+
+    @Test
+    public void NoProcessWhenOneDaySession() throws Exception {
+        //Arrange
+        ZonedDateTime timeDateStarted = ZonedDateTime.parse("2017-03-23T20:59:00.000+01:00[Europe/Warsaw]");
+        long sessionTime = 1258;
+
+        List<DailyUsage> dailyUsagesExpected = new ArrayList<>();
+        DailyUsage dailyUsage = new DailyUsage(timeDateStarted, sessionTime);
+        dailyUsagesExpected.add(dailyUsage);
+
+        //Act & Assert
+        List<DailyUsage> dailyUsagesActual = timeSpentCrossMidnightService.divideDaysWithTimeSessionCrossedMidnight(dailyUsagesExpected);
+
+        if (Objects.isNull(dailyUsagesActual) || dailyUsagesActual.isEmpty()) {
+            Assert.fail();
```

Comment:
Bad idea, if this would fail, I would have to debug the code, because:
1. I need first to navigate to this place in code in order to have any idea what is happening,
2. I would need to debug the code, becaue I would not know wheather dailyUsagesActual is empty, or is null.